js/ui/grid_core/ui.grid_core.editing.js

```diff
@@ -1673,14 +1700,35 @@ var EditingController = modules.ViewController.inherit((function() {
                 oldData = options.data,
                 rowKey = options.key,
                 $cellElement = $(options.cellElement),
-                editMode = getEditMode(that);
+                editMode = getEditMode(that),
+                deferred = new Deferred(),
+                setCellValueResult;
+
+            function createFailureHandler(deferred) {
+                return function(arg) {
+                    var error = arg instanceof Error ? arg : new Error(arg && String(arg) || "Unknown error");
```

Comment:
This line is duplicated. Try to refactor it. The createFailureHandler function from _saveEditDataCore method can be reused.